keel-core/src/main/kotlin/com/netflix/spinnaker/keel/persistence/memory/InMemoryArtifactRepository.kt

```diff
@@ -6,25 +6,36 @@ import com.netflix.spinnaker.keel.api.DeliveryArtifact
 import com.netflix.spinnaker.keel.api.DeliveryConfig
 import com.netflix.spinnaker.keel.persistence.ArtifactRepository
 import com.netflix.spinnaker.keel.persistence.NoSuchArtifactException
-import com.netflix.spinnaker.keel.persistence.sortAppVersion
 import org.slf4j.Logger
 import org.slf4j.LoggerFactory
 
 class InMemoryArtifactRepository : ArtifactRepository {
   private val artifacts = mutableMapOf<DeliveryArtifact, MutableList<ArtifactVersionAndStatus>>()
-  private val approvedVersions = mutableMapOf<Triple<DeliveryArtifact, DeliveryConfig, String>, MutableList<String>>()
-  private val deployedVersions = mutableMapOf<Triple<DeliveryArtifact, DeliveryConfig, String>, MutableList<String>>()
+  private val approvedVersions = mutableMapOf<Key, MutableList<String>>()
+  private val deployedVersions = mutableMapOf<Key, MutableList<String>>()
   private val log: Logger by lazy { LoggerFactory.getLogger(javaClass) }
 
+  private data class Key(
```

Comment:
I find `Triple` confusing to work with, so I created a `Key` class.

Reply:
Yeah, that's better